jwst/datamodels/model_base.py

```diff
@@ -79,6 +84,10 @@ def __init__(self, init=None, schema=None, extensions=None,
         kwargs: Aadditional arguments passed to lower level functions
         """
         # Set attributes used to hold information for asdf
+        if extensions is None:
+            extensions = [BaseExtension()]
+        else:
+            extensions.extend([BaseExtension()])
```

Comment:
That's an interesting question.  I must admit I don't know enough yet of how it works to really say.  I can investigate.

Reply:
I think you're right that we do not need it.  I.e. if I set `self._extensions = None` now, all the tests still pass.  Code delete-o-rama?

Reply:
I will hold off deleting the internal references to `self._extensions` because even though we don't use it, it does offer a way to have new extensions without having to register them through `entry_points`.  I will file a separate issue concerning `extensions` removal, and we can discuss its implications there.

Reply:
Removing the explicit use of extensions seems like the right thing to do given that all other packages are currently implementing asdf plugins.